### PR TITLE
feat: enable uninstalling PatchedMediaKeysApple

### LINKED ISSUE #4469: Add ability to uninstall PatchedMediaKeysApple polyfill
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Yes

**Is your feature request related to a problem? Please describe.**

Yes. We want to be able to install and uninstall [shaka.polyfill.PatchedMediaKeysApple](https://shaka-player-demo.appspot.com/docs/api/shaka.polyfill.PatchedMediaKeysApple.html) depending on the type of content we're playing, since we want to be able to swap between native and non-native Safari Fairplay playback. This is due to "startover" live content not being playable using Shaka Fairplay HLS (see https://github.com/shaka-project/shaka-player/issues/4431).

**Describe the solution you'd like**

A `shaka.polyfill.PatchedMediaKeysApple.uninstall()` method, that uninstalls the polyfill. Synchronously if possible.

**Describe alternatives you've considered**

We're currently using native Fairplay playback, but it has more quality issues that playing using Shaka, so we would like to use Shaka playback as much as possible.

One alternative solution is to fix #4431.

**Additional context**

-

### PR DESCRIPTION
**Pull request by martinstark**
Close #4469

The idea is to only enable uninstalling when it is desired, in order to not hold things in memory unnecessarily.

### CODE REVIEW

**Comment by martinstark** [comment]
Will re-open once I run some more tests.

**Comment by martinstark** [comment]
Re-opening to ask questions

File: lib/polyfill/patchedmediakeys_apple.js
Comment (martinstark) [inline lib/polyfill/patchedmediakeys_apple.js:42]: This does not work (cannot call get directly on prototype.mediaKeys), and I'm unsure if there's a need for saving and restoring the mediaKeys. Any input would be appreciatedb

**Review by avelad — APPROVED** [review]

**Comment by joeyparrish** [comment]
I don't understand this PR or the associated issue, and it seems really odd to me.  Let's discuss on the issue, but for now, please do not merge the PR.  Sorry for the inconvenience.

File: lib/polyfill/patchedmediakeys_apple.js
Comment (joeyparrish) [reply lib/polyfill/patchedmediakeys_apple.js:42]: If you need to backup and restore a getter, you may need to try getOwnPropertyDescriptor.  There is a good example of backing up and restoring a getter this way in test/player_unit.js.

Does this help?

File: lib/polyfill/patchedmediakeys_apple.js
Comment (joeyparrish) [inline lib/polyfill/patchedmediakeys_apple.js:81]: How difficult would it be to add a test that exercises this code?

**Review by joeyparrish — DISMISSED** [review]

**Comment by martinstark** [comment]
Thanks @joeyparrish, I'll look at adjusting according to your feedback and add a test

**Comment by martinstark** [comment]
@avelad @joeyparrish I've addressed the feedback, introduced unit tests and added a short blurb in the documentation. 

I opted for not restricting the unit test to only run on Safari. The functionality that's being tested is just overriding and restoring browser globals.

I don't know of a way of checking strict equality on the mediaKeys getter. The test does a best effort check to see if it looks like the value is no longer overridden, but it doesn't guarantee that it was restored.

I have not yet run the latest code updates on a Mac, I'll update the PR once I have.

**Comment by avelad** [comment]
@martinstark It seems that the tests have failed, can you check it?

**Comment by martinstark** [comment]
@avelad 
```bash
 1) cues ending exactly now
     TextDisplayer layout using browser-native rendering
     native-end-time-edge-case: Expected 0.9133498736570255 not to be less than 0.95.
```
I don't think this is related to my changes here, what do you think?

**Comment by avelad** [comment]
I'll re-run the failed jobs

**Comment by martinstark** [comment]
@avelad the safari test still fails, I've rebased on the lastest `main` to see if that makes a difference

**Comment by martinstark** [comment]
```bash
1) cues ending exactly now
     TextDisplayer layout using browser-native rendering
     native-end-time-edge-case: Expected 0.9133498736570255 not to be less than 0.95.
```
Keeps failing with exactly the same number. I've had a brief look at the test in question, as well as the text displayer, but no ideas spring to mind.

**Comment by avelad** [comment]
I'll ignore the fail test to approve the PR. Please send the final code please (no comments in the tests)!

**Comment by martinstark** [comment]
@avelad I tried to remove the unit tests to see if that's what was causing the Safari test suite failure. I've reverted this and squashed everything.

Thanks for your time!

File: docs/tutorials/fairplay.md
Comment (avelad) [inline docs/tutorials/fairplay.md:14]: This should be: `/* enableUninstall= */ true`. This keeps it consistent with the rest of the code.

**Review by avelad — CHANGES_REQUESTED** [review]
A minor nit, otherwise LGTM

File: docs/tutorials/fairplay.md
Comment (martinstark) [reply docs/tutorials/fairplay.md]: addressed

**Review by avelad — APPROVED** [review]

**Comment by martinstark** [comment]
@avelad not sure what your process is, I just squashed and force-pushed after you approval, but that dismisses your review. Any input?

**Review by avelad — APPROVED** [review]

**Comment by avelad** [comment]
The squash it's done by github, so really it's not a problem :) I'll approve again

**Comment by avelad** [comment]
@joeyparrish can you review it? Thanks!

**Pull request merged**
